### DISCUSSION TITLE
How can find library

### DISCUSSION THREAD

**Discussion by StanleyWu8787** [msys2/MINGW-packages#21912] [General]
Hi all,
Not sure it is good for this repo.
In current, I want to embed python code to C and the command is follow:
```
gcc -g embed.c -IC:/msys64/mingw64/include/python3.11  -LC:/msys64/mingw64/lib/python3.11/config-11 -lpython
```

But it can't find python library.
Does anyone know how to find the libray?

**Comment by Shriharsh-Deshmukh** [comment]
_**To solve the issue these steps you can give a try:**_

Looking for your Paths: Make sure that you specify the include path and the library correctly:

**Include path:** C:/msys64/mingw64/include/python3.11

**Library path:** C:/msys64/mingw64/lib/python3.11/config-11

Correcting the Linking Flag: Use the correct library name. C:: Consequently, overall, when applying code similar to the one followed, the commands are typically:

_Copy:_
gcc –g embed.c -IC:/msys64/mingw64/include/python3.11 -LC:/msys64/mingw64/lib/python3.11/config-11 -lpython3.11 

Instalation of Python Dev Files: Install Python dev headers by using MSYS2.

_Copy:_
pacman -S mingw-w64-x86_64-python3 

Set Environment Variables: Change to PYTHONHOME PYTHONPATH (if needed):

export PYTHONHOME=C:/msys64/mingw64

export PYTHONPATH=C:/msys64/mingw6A4/lib/python3.11

_Copy:_
Debuging: Let us append -v in order to check the directories where GCC looks for header files more precisely.

_Copy:_
gcc -g embed.c -IC:/msys64/mingw64/include/python3.11 -LC:/msys64/mingw64/lib/python3.11/config-11 -lpython3.11 -v 

This should help you to resolve this problem.

**Reply by Biswa96** [reply]
You can use pkgconf to get those paths.

```
$ pkgconf -cflags python3
-IC:/msys64/ucrt64/include/python3.11

$ pkgconf --keep-system-libs --libs python3
-LC:/msys64/ucrt64/lib -lpython3.11
```

**Reply by StanleyWu8787** [reply]
Thanks all,
I have resolve this issue.
Thanks a lot.
